### PR TITLE
Story/dfc 2295 lmi circuit breaker

### PR DESCRIPTION
**Pull request by trevorkapswarah**

### CODE REVIEW

File: DFC.Digital/DFC.Digital.AutomationTest.Utilities/SpecflowExtensions.cs [hunk 1]
```diff
@@ -114,5 +115,10 @@ private static string GetConditionalData(this TableRow item, string field, strin
         {
             return item.ContainsKey(field) ? item[field] : (defaultValue ?? string.Empty);
         }
+
+        private static T GetConditionalData<T>(this TableRow item, string field)
+        {
+            return item.ContainsKey(field) ? (T)(object)item[field] : default(T);
```
Comment (Muthuramana) [inline DFC.Digital/DFC.Digital.AutomationTest.Utilities/SpecflowExtensions.cs]: did you meant to write `return item.ContainsKey(field) ? (T)Convert.ChangeType(item[field], typeof(T)) : default;`

Comment (Muthuramana) [reply DFC.Digital/DFC.Digital.AutomationTest.Utilities/SpecflowExtensions.cs]: Also it makes sense to remove the string only func and change the signature of the new to 
```
private static T GetConditionalData<T>(this TableRow item, string field, T defaultValue = null)
        {
            return item.ContainsKey(field) ? (T)Convert.ChangeType(item[field], typeof(T)) : defaultValue ?? default;
```

File: DFC.Digital/DFC.Digital.Data/Model/JobProfileIndex.cs [hunk 2]
```diff
@@ -26,9 +26,13 @@ public class JobProfileIndex : IDigitalDataModel
...
 
         [IsFilterable, IsSortable, IsFacetable]
-        public string SalaryRange { get; set; }
+        public double SalaryStarter { get; set; }
+
+        [IsFilterable, IsSortable, IsFacetable]
+        public double SalaryExperienced { get; set; }
 
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1056:UriPropertiesShouldNotBeStrings", Justification = "This is an application field of type string, last segment and is not a complete uri")]
+        [IsFilterable, IsFacetable]
```
Comment (Muthuramana) [inline DFC.Digital/DFC.Digital.Data/Model/JobProfileIndex.cs]: Is it really facetable? UrlName is unique isn't it?

File: DFC.Digital/DFC.Digital.Web.Sitefinity.DfcSearchModule/SearchIndexEnhancers/JobProfileIndexEnhancer.cs [hunk 3]
```diff
@@ -46,18 +47,21 @@ public JobProfileIndex GetRelatedFieldsWithUrl(JobProfileIndex jobProfileIndex)
...
                 var salary = await salaryService.GetSalaryBySocAsync(JobProfile.SOCCode);
-                JobProfile.SalaryStarter = salaryCalculator.GetStarterSalary(salary);
-                JobProfile.SalaryExperienced = salaryCalculator.GetExperiencedSalary(salary);
+                jobProfileIndex.SalaryStarter = Convert.ToDouble(salaryCalculator.GetStarterSalary(salary));
+                jobProfileIndex.SalaryExperienced = Convert.ToDouble(salaryCalculator.GetExperiencedSalary(salary));
+            }
+            else
+            {
+                jobProfileIndex.SalaryStarter = Convert.ToDouble(JobProfile.SalaryStarter);
+                jobProfileIndex.SalaryExperienced = Convert.ToDouble(JobProfile.SalaryExperienced);
```
Comment (Muthuramana) [inline DFC.Digital/DFC.Digital.Web.Sitefinity.DfcSearchModule/SearchIndexEnhancers/JobProfileIndexEnhancer.cs]: do we need additional casting? isnt it possible to change the method signature or the model to match?

File: DFC.Digital/DFC.Digital.Web.Sitefinity.JobProfileModule/Mvc/Controllers/JobProfileDetailsController.cs [hunk 4]
```diff
@@ -74,7 +72,7 @@ public class JobProfileDetailsController : BaseJobProfileWidgetController
         public string MaxAndMinHoursAreBlankText { get; set; } = "Variable";
 
         [DisplayName("Hours time period")]
-        public string HoursTimePeriodText { get; set; } = "(per week)";
+        public string HoursTimePeriodText { get; set; } = "per week";
```
Comment (Muthuramana) [inline DFC.Digital/DFC.Digital.Web.Sitefinity.JobProfileModule/Mvc/Controllers/JobProfileDetailsController.cs]: This should not be changed, merge conflict?

File: DFC.Digital/DFC.Digital.Web.Sitefinity.DfcSearchModule/SearchIndexEnhancers/JobProfileIndexEnhancer.cs (hunk 3 above)
Comment (trevorkapswarah) [reply DFC.Digital/DFC.Digital.Web.Sitefinity.DfcSearchModule/SearchIndexEnhancers/JobProfileIndexEnhancer.cs]: Sitefinity defaults to decimal and Azure search requires double hence the conversion

**Pull request merged**